iroh/src/tls/verifier.rs

```diff
@@ -75,32 +72,35 @@ pub(super) struct ServerCertificateVerifier {
 ///
 /// or a raw public key.
 impl ServerCertificateVerifier {
-    pub(super) fn with_remote_peer_id(auth: Authentication, remote_peer_id: PublicKey) -> Self {
-        let mut trusted_spki = Vec::new();
-        let der_key = remote_peer_id
-            .public()
-            .to_public_key_der()
-            .expect("valid key");
-        let remote_key = SubjectPublicKeyInfoDer::from(der_key.into_vec());
-        trusted_spki.push(remote_key);
-
-        Self {
-            remote_peer_id,
-            auth,
-            trusted_spki,
-        }
+    pub(super) fn new(auth: Authentication) -> Self {
+        Self { auth }
     }
 }
 
+fn public_key_to_spki(remote_peer_id: &PublicKey) -> SubjectPublicKeyInfoDer<'static> {
+    let der_key = remote_peer_id
+        .public()
+        .to_public_key_der()
+        .expect("valid key");
+    let remote_key = SubjectPublicKeyInfoDer::from(der_key.into_vec());
+    remote_key
+}
+
 impl ServerCertVerifier for ServerCertificateVerifier {
     fn verify_server_cert(
         &self,
         end_entity: &Certificate,
         intermediates: &[Certificate],
-        _server_name: &rustls::pki_types::ServerName,
+        server_name: &rustls::pki_types::ServerName,
```

Comment:
who sets this value?

Reply:
That said - I find this reasoning quite convincing: The only place we use `tls::name` is in `Endpoint::connect`.
What we want to prevent is the server side setting the `server_name`. If that were the case, then there must've been some place in the sever-side code for us where we use `tls::name` to set the `server_name`. If the server name would've been set to something else, then `verify_sever_cert` would fail due to not parsing it correctly.
But there's no other occurrence of `tls::name` or greps for an `iroh.invalid` string.

FWIW